asserts/ifacedecls.go

```diff
@@ -309,8 +317,33 @@ func baseCompileConstraints(context string, cDef constraintsDef, target constrai
 		}
 		target.setAttributeConstraints(field, cstrs)
 	}
-	if defaultUsed == len(attributeConstraints)+len(idConstraints) {
-		return fmt.Errorf("%s must specify at least one of %s, %s", context, strings.Join(attrConstraints, ", "), strings.Join(idConstraints, ", "))
+	onClassic := cMap["on-classic"]
+	if onClassic == nil {
+		defaultUsed++
+	} else {
+		var c *OnClassicConstraint
+		switch x := onClassic.(type) {
+		case string:
+			switch x {
+			case "true":
+				c = &OnClassicConstraint{Classic: true}
+			case "false":
+				c = &OnClassicConstraint{Classic: false}
+			}
+		case []interface{}:
+			l, err := checkStringListInMap(cMap, "on-classic", fmt.Sprintf("on-classic in %s", context), validDistro)
+			if err != nil {
+				return err
+			}
+			c = &OnClassicConstraint{Classic: true, Distros: l}
```

Comment:
it's a bit of Go style impulse to go short, but I can move to lst if it's easier on the eye